independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -323,6 +323,17 @@ public InputStream getResourceAsStream(String unsanitisedName) {
             for (ClassPathElement i : elements) {
                 ClassPathResource res = i.getResource(name);
                 if (res != null) {
+                    if (res.isDirectory()) {
+                        try {
+                            return res.getUrl().openStream();
```

Comment:
Do we need to close the stream as well?

Reply:
No, given that this `getResourceAsStream` is meant to return a "live" stream, we return it live and expect the user/caller to close it whenever they are done using it (just like they would with regular non-directory based streams).

Reply:
Ah right... I didn't see which method this belonged to.

Reply:
Maybe related, but it just occurred to me that this method below may be broken? 

https://github.com/quarkusio/quarkus/blob/8b9c78882ae0a7aedf79c87dccbc73c81fe2f7eb/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/DirectoryClassPathElement.java#L97-L110

Reply:
Hello @gastaldi, do you mean it's broken because it isn't doing a check to see if the `file` is a directory, or something else?

Reply:
@jaikiran exactly

Reply:
The class name is misleading if it handles files too IMHO

Reply:
I think given that there's already an `isDirectory()` API on `ClassPathResource`, I think, it's the caller's responsibility to do that check before calling `getData()`.
This `getData()` could of course do checks for `isDirectory` but if it does find that it's an directory, then it either has to error out or open a stream on the URL represented by the file and then read from that. All doable, but I don't have the complete background around these classes, so I'm not sure if it should be done here in this method. @stuartwdouglas what are you thoughts?

Reply:
I think this is fine.